zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -559,9 +560,41 @@ impl Rw {
 
 impl From<&bus_mapping::circuit_input_builder::ExecStep> for ExecutionState {
     fn from(step: &bus_mapping::circuit_input_builder::ExecStep) -> Self {
-        // TODO: error reporting. (errors are defined in
-        // circuit_input_builder.rs)
-        debug_assert!(step.error.is_none());
+        if let Some(error) = step.error.as_ref() {
+            let state = match error {
+                ExecError::ExecutionReverted => ExecutionState::REVERT,
+                ExecError::InvalidOpcode => ExecutionState::ErrorInvalidOpcode,
+                ExecError::StackOverflow => ExecutionState::ErrorStackOverflow,
+                ExecError::StackUnderflow => ExecutionState::ErrorStackUnderflow,
+                ExecError::WriteProtection => ExecutionState::ErrorWriteProtection,
+                ExecError::Depth => ExecutionState::ErrorDepth,
+                ExecError::InsufficientBalance => ExecutionState::ErrorInsufficientBalance,
+                ExecError::ContractAddressCollision => {
+                    ExecutionState::ErrorContractAddressCollision
+                }
+                ExecError::InvalidCode => ExecutionState::ErrorInvalidCreationCode,
```

Comment:
I am not sure whether `ExecError::InvalidCode` is  corresonding to `ExecutionState::ErrorInvalidCreationCode`. 

@lispc @DreamWuGit do you think this line is correct?